src/sysext/sysext.c

```diff
@@ -91,6 +98,66 @@ static const struct {
         }
 };
 
+static bool need_reload(void) {
+        // Parse the mounted images to find out if we need
+        // to reload the daemon.
+        _cleanup_free_ char *f = NULL, *buf = NULL, **mounted_extensions = NULL;
+        int r;
+
+        STRV_FOREACH(p, arg_hierarchies) {
+            f = path_join(*p, image_class_info[arg_image_class].dot_directory_name, image_class_info[arg_image_class].short_identifier_plural);
+            if (!f)
+                    continue;
+
+            r = read_full_file(f, &buf, NULL);
+            if (r < 0)
+                    continue;
+
+            mounted_extensions = strv_split_newlines(buf);
+            if (!mounted_extensions)
+                    continue;
+
+            STRV_FOREACH(extension, mounted_extensions) {
+                    _cleanup_strv_free_ char **extension_release = NULL;
+                    const char *extension_reload_manager = NULL;
+                    r = load_extension_release_pairs(arg_root, arg_image_class, *extension, /* relax_extension_release_check */ true, &extension_release);
+                    if (r < 0) {
+                            log_debug_errno(r, "Failed to parse image %s extension-release metadata: %m", *extension);
```

Comment:
it's OK to ignore this error here, given this is foreign data, not our own. But please always say that we are ignoring it, i.e. say "ignoring" somewhere in the error: "Failed to parse extension-release medata data of %s, ignoring: %m".